util/chplenv/chpl_llvm.py

```diff
@@ -316,16 +339,32 @@ def get_gcc_prefix():
                         if os.path.isdir(inc):
                             gcc_prefix = mydir
 
-    return gcc_prefix.strip()
+        gcc_prefix = gcc_prefix.strip()
+        if gcc_prefix == '/usr':
+            # clang will be able to figure this out so don't
+            # bother with the argument here.
+            gcc_prefix = ''
+
+    return gcc_prefix
 
+# The bundled LLVM does not currently know to look in a particular Mac OS X SDK
+# so we provide a -isysroot arg to indicate which is used.
+#
+# Potential alternatives to -isysroot here include:
+#  * using the environment variable SDKROOT
+#  * providing a cmake argument to adjust the clang build
+#  * using -mmacosx-version-min=11.2 e.g.
+#
+# Additionally the -resource-dir arg indicates where to find some clang
+# dependencies.
+#
 # Returns a [ ] list of args
 @memoize
 def get_sysroot_resource_dir_args():
     args = [ ]
     target_platform = chpl_platform.get('target')
     llvm_val = get()
-    if (target_platform == "darwin" and
-        (llvm_val == "bundled" or llvm_val == "system")):
+    if (target_platform == "darwin" and llvm_val == "bundled"):
```

Comment:
This part of the PR message applies:

> It removes two workarounds I added in PR #18727 because reinstalling CommandLineTools fixed the problem for me and these workarounds weren't sufficient to get the third-party code compiling.

In other words I added it to solve a problem compiling on my system but the real solution was to fix the installation of some other software on my system. It shouldn't be necessary because the Homebrew clang should be able to compile a C program without such arguments.